### DISCUSSION TITLE
How to use convert back-and-forth with just one rates object?

### DISCUSSION THREAD

**Discussion by victorcarvalhosp** [dinerojs/dinero.js#762] [Q&A] [answered]
Hi everyone! First of all, thanks for the great job everyone is doing on Dinero 2.0. 

I'm confused about how can I use the convert function to convert back and forth between two different currencies. In the documentation, there's an [example showing how to convert from USD to EUR](https://v2.dinerojs.com/docs/api/conversions/convert) - but how can I convert back from EUR to USD without creating a new rates object?

I'm my application I have an object like this(let's assume these values are update daily): 

export const myRatesObject = {
  EUR: { amount: 95, scale: 2 },
  ETH: { amount: 64, scale: 5 },
  BTC: { amount: 3719503474549434, scale: 20 },
  BRL: { amount: 501, scale: 2 },
};

But this just works if I'm trying to convert these currencies to USD - but not the other way around. So for example, this works:

```
import { dinero, convert } from 'dinero.js';
import { USD, EUR } from '@dinero.js/currencies';

const d = dinero({ amount: 500, currency: USD });

convert(d, EUR, myRatesObject); // return EUR dinero object
```

But I expect the other way around to work as well - without the need of creating a new rates object for each currency, something like:

```
import { dinero, convert } from 'dinero.js';
import { USD, EUR } from '@dinero.js/currencies';

const d = dinero({ amount: 500, currency: EUR });

convert(d, USD, myRatesObject); // I expect it to return an USD dinero object - just using the same rates object
```

If I need to create a new rates object for each currency, then these objects become unmanageable as I can have almost infinite combinations. I want to have just one single source of truth, that is the USD rates object and be able to convert everything back and forth just using it. 

I'm missing something here? Can someone clear my mind or give some examples of how can I achieve this?

**Comment by sarahdayan** [comment, marked as answer]
Hey @victorcarvalhosp! The `convert` function works one-way by design: you provide rates relative to the source currency. If your rates object is USD-based (all rates expressed as "1 USD = X"), you can only convert _from_ USD.

To convert between two non-base currencies (e.g. EUR to BRL), you'd need to go through the base currency. For example, convert EUR to USD first using EUR-based rates, then USD to BRL using your USD-based rates. Alternatively, you can compute cross rates on the fly before passing them to `convert`.

This is consistent with how exchange rate APIs work. They give you rates relative to a base, and cross-rate conversion requires an intermediate step.
